### PR TITLE
evm: add public getters

### PR DESCRIPTION
**Pull request by panoel**
Add some useful external facing getters.

### CODE REVIEW

File: evm/src/TransceiverRegistry.sol [hunk 1]
```diff
@@ -416,13 +416,23 @@ abstract contract TransceiverRegistry {
...
     /// @return result The registered transceivers for the given integrator.
     function getTransceivers(address integrator) external view returns (address[] memory result) {
         result = _getRegisteredTransceiversStorage()[integrator];
     }
 
+    /// @notice Returns the queried transceiver addresses.
+    /// @param integrator The integrator address
+    /// @param index The index into the integrator's transceivers array
+    /// @return result The registered transceiver address.
+    function getTransceiverByIndex(address integrator, uint8 index) external view returns (address result) {
```
[the remark this answers is not in the document]
Comment (panoel) [reply evm/src/TransceiverRegistry.sol:430]: Added

File: evm/src/Router.sol [hunk 2]
```diff
@@ -208,6 +208,14 @@ contract Router is IRouterAdmin, IRouterIntegrator, IRouterTransceiver, MessageS
         return integratorConfigs[integrator].admin;
     }
 
+    /// @notice Returns the pending_admin for a given integrator.
+    /// @param integrator The address of the integrator contract.
+    /// @return address The address of the administrator contract
```
Comment (bruce-riley) [inline evm/src/Router.sol:213]: "pending administrator contract"?

File: evm/src/TransceiverRegistry.sol [hunk 3]
```diff
@@ -416,13 +416,40 @@ abstract contract TransceiverRegistry {
...
-    /// @notice Returns the registered send side transceiver addresses for the given integrator.
+    /// @notice Returns all the registered transceiver addresses for the given integrator.
     /// @param integrator The integrator address
     /// @return result The registered transceivers for the given integrator.
     function getTransceivers(address integrator) external view returns (address[] memory result) {
         result = _getRegisteredTransceiversStorage()[integrator];
     }
 
+    /// @notice Returns the queried transceiver addresses.
+    /// @param integrator The integrator address
```
Comment (bruce-riley) [inline evm/src/TransceiverRegistry.sol:427]: Extreme nit: some of the prologue lines end with a period and some don't.

**Review by bruce-riley — DISMISSED** [review]

File: evm/src/Router.sol (hunk 2 above)
Comment (panoel) [reply evm/src/Router.sol:213]: Fixed

File: evm/src/TransceiverRegistry.sol (hunk 3 above)
Comment (panoel) [reply evm/src/TransceiverRegistry.sol:427]: Maybe just the sentences should have periods and the clauses shouldn't

Comment (panoel) [reply evm/src/TransceiverRegistry.sol:427]: Fixed

File: evm/src/TransceiverRegistry.sol [hunk 4]
```diff
@@ -416,17 +416,44 @@ abstract contract TransceiverRegistry {
...
+    }
+
+    // =============== PUBLIC GETTERS ========================================
+
+    /// @notice Returns the queried transceiver's index.
+    /// @param integrator The integrator address
+    /// @param transceiver The address of this transceiver
+    /// @return result The registered transceiver index
+    function getTransceiverIndex(address integrator, address transceiver) external view returns (uint8 result) {
+        uint256 length = _getRegisteredTransceiversStorage()[integrator].length;
```
Comment (evan-gray) [inline evm/src/TransceiverRegistry.sol:441]: this should simply use `_getTransceiverInfosStorage`, no need for looping

File: evm/src/TransceiverRegistry.sol [hunk 5]
```diff
@@ -227,7 +227,7 @@ abstract contract TransceiverRegistry {
         return transceiverInfos[integrator][transceiver].index;
     }
 
-    /// @dev It is assumed that the integrator address is already validated (and not 0)
+    /// @dev It is assumed that the integrator address is already validated (and not 0).
```
Comment (evan-gray) [inline evm/src/TransceiverRegistry.sol:230]: what did the periods do to you? :P

**Review by evan-gray — CHANGES_REQUESTED** [review]

File: evm/src/TransceiverRegistry.sol (hunk 5 above)
Comment (panoel) [reply evm/src/TransceiverRegistry.sol:230]: I blame Bruce.

**Review by evan-gray — APPROVED** [review]

**Review by bruce-riley — APPROVED** [review]

**Pull request merged**